eZ/Publish/SPI/Search/Indexer.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\Search;
+
+/**
+ * Indexer for handlers that index stuff.
+ */
+interface Indexer
```

Comment:
`IndexUpdating`? Too bad we can't have abstract interfaces ;)

> Also unsure why content (fulltext, full content domain) would care about location deletion.

Because it can result in deletion of Content.
The same for Content deletion, it needs to remove all Content's Locations.